### PR TITLE
Address mypy and pylint issues in `anomaly_classification`

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Coverage

image

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,56 +1,75 @@
 default_language_version:
```
Comment (samet-akcay) [inline .pre-commit-config.yaml:1]: This formatting change is by the `prettier`, right?

File: external/anomaly/anomaly_classification/openvino.py [hunk 2]
```diff
@@ -133,51 +133,42 @@ def optimize(
         with tempfile.TemporaryDirectory() as tempdir:
             xml_path = os.path.join(tempdir, "model.xml")
             bin_path = os.path.join(tempdir, "model.bin")
-            with open(xml_path, "wb") as xml_file:
-                xml_file.write(self.task_environment.model.get_data("openvino.xml"))
-            with open(bin_path, "wb") as bin_file:
-                bin_file.write(self.task_environment.model.get_data("openvino.bin"))
```
Comment (samet-akcay) [inline external/anomaly/anomaly_classification/openvino.py:139]: tbh, I'd prefer the old approach. When reading, I find it flowing better.

File: external/anomaly/anomaly_classification/openvino.py [hunk 3]
```diff
@@ -133,51 +133,42 @@ def optimize(
...
+            with open(xml_path, "wb") as weight_file:
+                weight_file.write(self.task_environment.model.get_data("openvino.xml"))
+            with open(bin_path, "wb") as weight_file:
+                weight_file.write(self.task_environment.model.get_data("openvino.bin"))
 
-            model_config = ADDict({"model_name": "openvino_model", "model": xml_path, "weights": bin_path})
-
-            model = load_model(model_config)
+            model = load_model(ADDict({"model_name": "openvino_model", "model": xml_path, "weights": bin_path}))
```
Comment (samet-akcay) [inline external/anomaly/anomaly_classification/openvino.py:141]: same here.

`model = load_model(ADDict({"model_name": "openvino_model", "model": xml_path, "weights": bin_path}))` would be hard to follow. Maybe, we could remove the line break between `model_config` and `model` to keep them together, in terms of consistency.

```
model_config = ADDict({"model_name": "openvino_model", "model": xml_path, "weights": bin_path})
model = load_model(model_config)
```

File: external/anomaly/setup.py [hunk 4]
```diff
@@ -1,4 +1,7 @@
-#!/usr/bin/env python
+"""
+Install anomalib wrapper for OTE
+"""
+
```
Comment (samet-akcay) [inline external/anomaly/setup.py:4]: Should we also add the license header as well

File: external/anomaly/anomaly_classification/openvino.py [hunk 5]
```diff
@@ -133,51 +133,42 @@ def optimize(
...
-            with open(xml_path, "wb") as xml_file:
-                xml_file.write(self.task_environment.model.get_data("openvino.xml"))
-            with open(bin_path, "wb") as bin_file:
-                bin_file.write(self.task_environment.model.get_data("openvino.bin"))
+            with open(xml_path, "wb") as weight_file:
+                weight_file.write(self.task_environment.model.get_data("openvino.xml"))
+            with open(bin_path, "wb") as weight_file:
+                weight_file.write(self.task_environment.model.get_data("openvino.bin"))
 
-            model_config = ADDict({"model_name": "openvino_model", "model": xml_path, "weights": bin_path})
```
Comment (samet-akcay) [inline external/anomaly/anomaly_classification/openvino.py:141]: I'm looking at the documentation of `addict`, where a dictionary is created by `Dict` itself. If we don't use the native `Dict` in the module, I think we could also do the same. It's because when reading the code `ADDict` could  distracts.

File: external/anomaly/anomaly_classification/openvino.py [hunk 6]
```diff
@@ -133,51 +133,42 @@ def optimize(
...
 
+        compressed_model = create_pipeline(algorithms, engine).run(model)
         compress_model_weights(compressed_model)
 
         with tempfile.TemporaryDirectory() as tempdir:
             save_model(compressed_model, tempdir, model_name="model")
-            with open(os.path.join(tempdir, "model.xml"), "rb") as xml_file:
-                output_model.set_data("openvino.xml", xml_file.read())
-            with open(os.path.join(tempdir, "model.bin"), "rb") as bin_file:
-                output_model.set_data("openvino.bin", bin_file.read())
```
Comment (samet-akcay) [inline external/anomaly/anomaly_classification/openvino.py:180]: same here, I'd prefer the old approach in terms of readability.

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Couple of minor comments. Thanks for fixing these, looks a lot better now.

File: external/anomaly/anomaly_classification/openvino.py (hunk 2 above)
Comment (ashwinvaidya17) [reply external/anomaly/anomaly_classification/openvino.py:139]: I prefer it as well but I was trying to resolve `too many locals` issue. But I think it would be better to disable it as it comes at the expense of readability.

Comment (samet-akcay) [reply external/anomaly/anomaly_classification/openvino.py:139]: Since we have the below 4 times in the code
```
with open(xml_path, "wb") as xml_file:
                xml_file.write(self.task_environment.model.get_data("openvino.xml"))
```

We could then maybe define a function in `optimize` method such as:
```
def __save_weights(path: str, data: byte?) -> None:
    with open(path, "wb") as file:
        file.write(data)
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ashwinvaidya17) [reply .pre-commit-config.yaml:1]: Yes

**Comment by ashwinvaidya17** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
This branch has conflicts that must be resolved
Use the web editor or the  to resolve conflicts.
Conflicting files
.pre-commit-config.yaml

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by ashwinvaidya17** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Review by Ilya-Krylov — APPROVED** [review]

**Review by samet-akcay — APPROVED** [review]

**Review by samet-akcay — APPROVED** [review]
lgtm
